python/cugraph/structure/graph.py

```diff
@@ -493,6 +533,11 @@ def number_of_edges(self, directed_edges=False):
         Get the number of edges in the graph.
 
         """
+        if self.distributed:
+            if self.edgelist is not None:
+                return len(self.edgelist.edgelist_df)
+            else:
```

Comment:
If an adj list is present I assume we could just get it from there too.

Reply:
Currently we don't allow initialization of distributed graphs using adjlist. So for a distributed graph, if it is not empty,  would mean an edgelist is always present.